src/FormField/CheckBox.php

```diff
@@ -13,13 +13,26 @@ class CheckBox extends Generic
 
     public $defaultTemplate = 'formfield/checkbox.html';
 
-    public $label;
+    /**
+     * Label appears to the right of the checkbox. If label is not set specifically
+     * then the $caption property will be displayed as a label instead.
+     */
+    public $label = null;
+
+    public function __construct($label = null, $class = null)
```

Comment:
with constructors it's normal not to call parent. The idea of a constructor is to simply set some default values. It should not be complex. We do not support array argument on purpose.